### PR TITLE
[stdlib]Catch the reasonable mistake 'UnsafeRawMutableBufferPointer'

### PR DESCRIPTION
**Pull request by devKobe24**
Issue: [[SR-6360](https://bugs.swift.org/browse/SR-6360)]
Tag: https://github.com/apple/swift/issues/48910

- Modified: stdlib/public/core/UnsafeRawBufferPointer.swift
- Modified: stdlib/public/core/UnsafeRawBufferPointer.swift.gyb

Resolve: https://github.com/apple/swift/issues/48910, fix: https://github.com/apple/swift/pull/69909

### CODE REVIEW

File: stdlib/public/core/UnsafeRawBufferPointer.swift
Comment (glessard) [inline stdlib/public/core/UnsafeRawBufferPointer.swift:1]: We shouldn't add this file.

File: stdlib/public/core/UnsafeRawBufferPointer.swift.gyb [hunk 1]
```diff
@@ -11,7 +11,8 @@
 //===----------------------------------------------------------------------===//
 
 %import gyb
-
+@available(*, unavailable, renamed: "UnsafeMutableRawBufferPointer")
+typealias UnsafeRawMutableBufferPointer = UnsafeMutableRawBufferPointer
```
Comment (glessard) [inline stdlib/public/core/UnsafeRawBufferPointer.swift.gyb:15]: These are the lines we need. It's preferable not to make whitespace changes in the rest of the file alongside a real change such as this.

File: test/stdlib/UnsafePointerDiagnostics.swift
Comment (glessard) [inline test/stdlib/UnsafePointerDiagnostics.swift:1]: Please add a test in this file that checks the expected renaming suggestion is emitted at compile time.

**Review by glessard — CHANGES_REQUESTED** [review]
Thanks for trying again @devKobe24! This needs some cleanup and also a test.

File: stdlib/public/core/UnsafeRawBufferPointer.swift
Comment (devKobe24) [reply stdlib/public/core/UnsafeRawBufferPointer.swift:1]: Okay, I will delete this file.

File: stdlib/public/core/UnsafeRawBufferPointer.swift.gyb (hunk 1 above)
Comment (devKobe24) [reply stdlib/public/core/UnsafeRawBufferPointer.swift.gyb:15]: Okay, I will restore the whitespace changes in the file to their original state.

File: test/stdlib/UnsafePointerDiagnostics.swift
Comment (devKobe24) [reply test/stdlib/UnsafePointerDiagnostics.swift:1]: Okay, I will add a test to the file to check if the expected renaming suggestion is emitted at compile time.

**Pull request closed without merging**